vmdb/app/models/miq_worker.rb

```diff
@@ -88,7 +88,7 @@ def self.server_scope(server_id = nil)
       server = MiqServer.my_server
       server_id = server.id unless server.nil?
     end
-    where(:miq_server_id => server_id)
+    where(:miq_server_id => server_id, :type => self.to_s)
```

Comment:
1. do you need to add `:type => ...` to the `current_scope` guard clause?

2. Rails typically adds `:type` behind the scenes. Surprised you'd have to add it.
So if you're not seeing `:type` in the query, I feel this fix is good.